### PR TITLE
fix: allow GH priv key from env in helm chart

### PR DESCRIPTION
**Pull request by apr-1985**
Allowing the ability to get the GH priv key from the env when using the helm chart.
Feature Code added in https://github.com/actions-runner-controller/actions-runner-controller/pull/383

I have tested switching between Env and file by toggling the new helm chart variable and this works without issue.

This feature increases the interop with secrets managers that inject to ENV at run time.

### CODE REVIEW

**Comment by apr-1985** [comment]
Is there anything else I need to do get this reviewed please?

**Comment by marchenkov** [comment]
I have the same issue - we are using vault and all secrets must be passed as ENV (not as file `/etc/actions-runner-controller/github_app_private_key`).
Is there any chance to review/merge this PR?
Thanks

**Comment by mumoshu** [comment]
We're going to review and merge this before our next controller minor release! Thanks for your patience.

**Comment by mumoshu** [comment]
@marchenkov Hey! How does this PR solve your issue?

> we are using vault and all secrets must be passed as ENV

If you're using [vault-k8s' injector](https://github.com/hashicorp/vault-k8s) you should get files instead of envvars.
How are you integrating vault with your setup?

**Comment by apr-1985** [comment]
> We're going to review and merge this before our next controller minor release! Thanks for your patience.

Many thanks mainly commented to stop the stale bot 😀

**Comment by marchenkov** [comment]
> @marchenkov Hey! How does this PR solve your issue?
> 
> > we are using vault and all secrets must be passed as ENV
> 
> If you're using [vault-k8s' injector](https://github.com/hashicorp/vault-k8s) you should get files instead of envvars. How are you integrating vault with actions-runner-controller in your setup?

I'm use vault-secrets-webhook and no way to inject to file.
https://banzaicloud.com/docs/bank-vaults/mutating-webhook/configuration/
So this PR can solve my problem, because I can read PEM from secret (now I receive error that file `/etc/actions-runner-controller/github_app_private_key` not contain PEM cert).

**Comment by mumoshu** [comment]
@marchenkov I was also wondering if this hard-coded env name and secretKeyRef values might affect your use-case

```
key: github_app_private_key
name: {{ include "actions-runner-controller.secretName" . }}
```

But after reading how bank-vaults mutating webhook works, perhaps that's also no issue.

Thanks for confirming!

File: charts/actions-runner-controller/templates/deployment.yaml [hunk 1]
```diff
@@ -84,7 +84,15 @@ spec:
...
         - name: GITHUB_APP_PRIVATE_KEY
+          {{- if .Values.authSecret.github_private_key_from_disk }}
           value: /etc/actions-runner-controller/github_app_private_key
+          {{- else }}
+          valueFrom:
+            secretKeyRef:
+              key: github_app_private_key
+              name: {{ include "actions-runner-controller.secretName" . }}
+              optional: true
+          {{- end }}
```
Comment (mumoshu) [inline charts/actions-runner-controller/templates/deployment.yaml:95]: @apr-1985 @marchenkov @toast-gear Can we just do this instead, without the extra flag?

```suggestion
          valueFrom:
            secretKeyRef:
              key: github_app_private_key
              name: {{ include "actions-runner-controller.secretName" . }}
              optional: true
```

Regardless of we use the local file or the secretKeyRef, the content of the private key is loaded from `controller-manager` secret anyway. Then, this won't be a breaking change, right?

**Comment by mumoshu** [comment]
I'm now wondering if `authSecret.enabled=false` added via #937 with the existing `env` helm value can be an alternative to your use-case @apr-1985 @marchenkov WDYT?

File: charts/actions-runner-controller/templates/deployment.yaml (hunk 1 above)
Comment (marchenkov) [reply charts/actions-runner-controller/templates/deployment.yaml:95]: @mumoshu, I can confirm that your variant works for me too. My main problem - that file `/etc/actions-runner-controller/github_app_private_key` not found. So both options are valid for me, but your proposal the better. Thanks

Comment (mumoshu) [reply charts/actions-runner-controller/templates/deployment.yaml:95]: Thanks for your confirmation! Let's go ahead without the if block then.

File: charts/actions-runner-controller/values.yaml [hunk 2]
```diff
@@ -23,10 +23,12 @@ enableLeaderElection: true
 
 # Only 1 authentication method can be deployed at a time
 # Uncomment the configuration you are applying and fill in the details
+# Setting github_private_key_from_disk to false will load the key from ENV
 authSecret:
   enabled: true
   create: false
   name: "controller-manager"
+  github_private_key_from_disk: true
```
Comment (mumoshu) [inline charts/actions-runner-controller/values.yaml:31]:
```suggestion
```

File: charts/actions-runner-controller/values.yaml [hunk 3]
```diff
@@ -23,10 +23,12 @@ enableLeaderElection: true
 
 # Only 1 authentication method can be deployed at a time
 # Uncomment the configuration you are applying and fill in the details
+# Setting github_private_key_from_disk to false will load the key from ENV
```
Comment (mumoshu) [inline charts/actions-runner-controller/values.yaml:26]:
```suggestion
#
# If authSecret.enabled=true these values are inherited to actions-runner-controller's controller-manager container's env.
#
# Do set authSecret.enabled=false and set env if you want full control over
# the GitHub authn related envvars of the container.
# See https://github.com/actions-runner-controller/actions-runner-controller/pull/937 for more details.
```

**Review by mumoshu — APPROVED** [review]
Merging. Thanks for your contribution and support @apr-1985 @marchenkov ☺️

**Pull request merged**